PACKAGE = "team_triangle"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# param, data_type, val, desc, default, min, max
gen.add("speed", double_t, 0, "controls the speed of the robot", 0.6, 0.0, 10.0)
gen.add("thresh", int_t, 0, "threshold for detecting the line", 211, 0, 255)
gen.add("offset_turn", double_t, 0, "controls the turning angle of the robot", 3.0, 1.0, 6.0)
gen.add("enable_drive", bool_t, 0, "sets the robot in motion", False)

gen.add("hue_l", double_t, 0, "hue low",  50.0, 0, 255)
gen.add("hue_h", double_t, 0, "hue high", 255.0, 0, 255)

gen.add("sat_l", double_t, 0, "sat low",  0.0, 0, 255)
gen.add("sat_h", double_t, 0, "sat high", 255.0, 0, 255)

gen.add("val_l", double_t, 0, "val low",  240.0, 0, 255)
gen.add("val_h", double_t, 0, "val high", 255.0, 0, 255)

# the 3rd parameter must be this file's name
exit(gen.generate(PACKAGE, "follow_line", "FollowLine"))
